tests/basic.spec.ts

```diff
@@ -0,0 +1,86 @@
+import { test, expect } from './fixtures/base';
+import { fillPopup } from './pages/popup';
+import { getMessage } from './helpers';
+
+test.beforeEach(async ({ popup }) => {
+  await popup.reload();
+});
+
+test('should load popup', async ({ popup }) => {
+  await popup.bringToFront();
+  await expect(popup).toHaveTitle('Web Monetization Extension');
+  await expect(popup.locator('#popup-container')).toBeAttached();
+  await expect(popup.locator('header')).toHaveText('Web Monetization');
+  await expect(popup.locator('header img')).toHaveAttribute(
+    'src',
+    /logo\.svg$/,
+  );
+});
+
+test('shows connect form if not connected', async ({ page, popup }) => {
+  await page.goto('https://example.com');
+
+  await expect(popup).toHaveTitle('Web Monetization Extension');
+  await expect(popup.locator('#popup-container')).toBeAttached();
+  await expect(popup.locator('header')).toHaveText('Web Monetization');
+
+  await expect(popup.locator('form')).toBeVisible();
+  await expect(popup.locator('form button[type="submit"]')).toBeVisible();
+  await expect(popup.locator('form button[type="submit"]')).toHaveText(
+    'Connect',
+  );
+});
+
+test.describe('should fail to connect if:', () => {
+  test('invalid URL provided', async ({ background, popup }) => {
+    const connectButton = await fillPopup(popup, {
+      walletAddressUrl: 'abc',
+      amount: '10',
+      recurring: false,
+    });
+    await connectButton.click();
+    await expect(popup.locator('p.text-error')).toHaveText('Failed to fetch');
+    expect(
+      await background.evaluate(() => {
+        return chrome.storage.local.get(['connected']);
+      }),
+    ).toEqual({ connected: false });
+  });
+
+  test('invalid wallet address provided', async ({ background, popup }) => {
+    const connectButton = await fillPopup(popup, {
+      walletAddressUrl: 'https://example.com',
+      amount: '10',
+      recurring: false,
+    });
+    await expect(popup.locator('p.text-error')).toContainText(
+      'Invalid wallet address.',
+    );
+
+    await connectButton.click();
+    await expect(popup.locator('p.text-error')).toContainText(
+      'Unexpected token',
```

Comment:
We should probably update these messages.

Reply:
Afterwards as a separate PR (for https://github.com/interledger/web-monetization-extension/issues/565), yes.